libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -143,7 +143,19 @@ func (p *addHostMetadata) loadData() error {
 		return err
 	}
 
-	data := host.MapHostInfo(features.FQDN(), h.Info())
+	hostname := h.Info().Hostname
+	if features.FQDN() {
+		fqdn, err := h.FQDN()
+		if err != nil {
+			// FQDN lookup is "best effort".  We log the error, fallback to
+			// the OS-reported hostname, and move on.
+			p.logger.Warnf("unable to lookup FQDN: %s, using hostname = %s as FQDN", err.Error(), hostname)
```

Comment:
How often would this be logged? We may want to lower this to Debug to avoid unnecessarily spamming the logs when this fails.

Reply:
It feels like we would be better off with a metrics counter for how often this resolution has failed.

That would tell us it's failing without worrying about log spam, and we could then turn on debug logging to see exactly what is happening.

Reply:
> It feels like we would be better off with a metrics counter for how often this resolution has failed.
> 
> That would tell us it's failing without worrying about log spam, and we could then turn on debug logging to see exactly what is happening.

Makes sense.  Let me look into implementing a metrics counter.